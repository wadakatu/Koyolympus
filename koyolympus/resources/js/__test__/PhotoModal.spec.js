import {shallowMount} from "@vue/test-utils";
import Component from '@/PhotoModalComponent.vue';

let wrapper;
beforeEach(() => {
    wrapper = shallowMount(Component, {
        propsData: {
            val: {
                id: "10",
                url: '/test',
            }
        },
    });
});

describe('Testing computed', () => {
    test('Testing get photoId', () => {
        expect(wrapper.vm.photoId).toBe('10');
        expect(wrapper.vm.photoId).not.toBe('11');
    });
});

describe('Testing v-bind', () => {
    test('Testing src in img tag', () => {
        expect(wrapper.find('img').attributes().src).toBe('/test');
    });
    test('Testing id in like-component', () => {
        expect(wrapper.find('like-component-stub').attributes().id).toBe('10');
    });
});

describe('Testing @event', () => {
    test('emit event fire when clicked div tag(#overlay)', () => {
        wrapper.find('#overlay').trigger('click.self');
        expect(wrapper.emitted('close')).not.toBeUndefined();
    });
    test('emit event fire when clicked div tag(#modal-content)', () => {
        wrapper.find('#modal-content').trigger('click.self');
        expect(wrapper.emitted('close')).not.toBeUndefined();
    });
    test('emit event DO NOT fire when clicked div tag(#modal-content-top)', () => {
        wrapper.find('#modal-content-top').trigger('click.self');
        expect(wrapper.emitted('close')).toBeUndefined();
    });
    test('emit event DO NOT fire when clicked div tag(#modal-content-bottom)', () => {
        wrapper.find('#modal-content-bottom').trigger('click.self');
        expect(wrapper.emitted('close')).toBeUndefined();
    });
});
