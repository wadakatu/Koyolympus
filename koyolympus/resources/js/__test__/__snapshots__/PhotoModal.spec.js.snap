// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SnapShot already liked SnapShot 1`] = `
<transition-stub
  name="component-fade"
>
  <div
    id="overlay"
  >
    <div
      id="modal-content"
    >
      <div
        id="modal-content-top"
      >
        <img
          alt="This photo taken by Koyo Isono."
          src="/test"
        />
      </div>
       
      <div
        id="modal-content-bottom"
      >
        <button
          class="static"
          id="like-heart"
        />
         
        <p
          id="like-count"
        >
          Like：0
        </p>
      </div>
    </div>
  </div>
</transition-stub>
`;

exports[`Testing SnapShot not liked SnapShot 1`] = `
<transition-stub
  name="component-fade"
>
  <div
    id="overlay"
  >
    <div
      id="modal-content"
    >
      <div
        id="modal-content-top"
      >
        <img
          alt="This photo taken by Koyo Isono."
          src="/test"
        />
      </div>
       
      <div
        id="modal-content-bottom"
      >
        <button
          class=""
          id="like-heart"
        />
         
        <p
          id="like-count"
        >
          Like：0
        </p>
      </div>
    </div>
  </div>
</transition-stub>
`;

exports[`Testing SnapShot push button and liked 1`] = `
<transition-stub
  name="component-fade"
>
  <div
    id="overlay"
  >
    <div
      id="modal-content"
    >
      <div
        id="modal-content-top"
      >
        <img
          alt="This photo taken by Koyo Isono."
          src="/test"
        />
      </div>
       
      <div
        id="modal-content-bottom"
      >
        <button
          class="press"
          id="like-heart"
        />
         
        <p
          id="like-count"
        >
          Like：0
        </p>
      </div>
    </div>
  </div>
</transition-stub>
`;
